IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m

```diff
@@ -111,16 +166,36 @@ - (NSString *)telemetryString
     return result;
 }
 
-- (NSArray<MSIDRequestTelemetryErrorInfo *> *)errorsInfo
+#pragma mark - NSSecureCoding
+
+#define kSchemaVersion              @"schemaVersion"
+#define kSilentSuccessfulCount      @"silentSuccessfulCount"
+#define kErrorsInfo                 @"errorsInfo"
+
+- (void)encodeWithCoder:(NSCoder *)encoder
 {
-    __block NSArray *errorsInfoCopy;
-    dispatch_sync(self.synchronizationQueue, ^{
-        errorsInfoCopy = [NSArray arrayWithArray:_errorsInfo];
-    });
-    return errorsInfoCopy;
+    [encoder encodeInteger:_schemaVersion forKey:kSchemaVersion];
+    [encoder encodeInteger:_silentSuccessfulCount forKey:kSilentSuccessfulCount];
+    [encoder encodeObject:_errorsInfo forKey:kErrorsInfo];
 }
 
-#pragma mark - Private
+- (instancetype)initWithCoder:(NSCoder *)decoder
+{
+    NSInteger schemaVersion = [decoder decodeIntegerForKey:kSchemaVersion];
+    NSInteger silentSuccessfulCount = [decoder decodeIntegerForKey:kSilentSuccessfulCount];
+    
+    NSSet *classes = [NSSet setWithObjects:[NSMutableArray class], [MSIDRequestTelemetryErrorInfo class], nil];
```

Comment:
I remember we had a crash on iOS 11.2 around similar logic. Can you double check it wouldn't be a problem here? https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/677

Reply:
Looks like the code change in the linked PR has to do with creating a NSSet of data structures within the archived dictionary. I have the same structure in this PR, so I'm hoping it should be good. The simulators only go back to 11.4 from what I can see on XCode, so I'm not sure if I can test it.